File: Java_Generated/Java/Maven/src/main/java/external/trading/database/interfaces/ITradingDatabaseHistoryInterface.java
package external.trading.database.interfaces;

import cancellation.interfaces.ICancellation;
import external.trading.database.classes.HistoricalDataMessageDateTime;

import java.util.Map;
import java.util.concurrent.CompletableFuture;

public interface ITradingDatabaseHistoryInterface {

    CompletableFuture<Map<String, Object>> getSymbolsAsync(ICancellation cancellation);
    CompletableFuture<HistoricalDataMessageDateTime[]> getHistoricalDataMessageDateTimesAsync(Object id, double begin, double end, ICancellation cancellation);

}

File: src/Web/AspireTradingApp/frontend/src/ExternalObjects/Trading/Interfaces/ITradingDatabaseHistoryInterface.ts
import type { HistoryMessage } from "../Database/HistoryMessage";

export interface ITradingDatabaseHistoryInterface {

    getSymbolsAsync(): Promise<string[][]>

    getHistoricalDataMessageDateTimesAsync(id: any, period: string,  symbol: string, begin: number,
        end: number, cancellation: AbortController): Promise<HistoryMessage[]>;

}